numba/core/typing/templates.py

```diff
@@ -876,6 +885,26 @@ def get_template_info(self):
         }
         return info
 
+    @functools.lru_cache()
+    def get_cache_deps_info(self, sig):
```

Comment:
Could you add type annotation to this function?

Reply:
adding a type annotation to `sig` can be done in 2 ways, tell me what you think:
- duplicating the `SignatureLike` typealias from caching.py (duplicating it because it cannot be imported)
- directly writing `pt.Union[Signature, pt.Tuple[types.Type, ...], str]` in the function declaration, without alias.

I'll go with the first, but let me know if you think the second is better.